### PR TITLE
docs: Add brief migration guide

### LINKED ISSUE #252: Migration to non-legacy bots
The rtm integration is now legacy. See #251. What do we need to do to un-legacy a slack-ruby-bot?

### PR DESCRIPTION
**Pull request by wasabigeek**
Fixes #252.

I may not have the opportunity to verify this flow for some time, if anyone has tried it, do let me know if it works / there are any issues!

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Migrating from Legacy API Tokens
+
+New Slack Apps require authentication via OAuth, returning an access token that should be a drop-in replacement for `SLACK_API_TOKEN`. As of the time of writing, the tokens never expire, so you could potentially write your own OAuth flow to retrieve the access token separately. Alternatively, you could migrate to [slack-ruby-bot-server](https://github.com/slack-ruby/slack-ruby-bot-server).
+
+## Migrating to slack-ruby-bot-server
+Overview:
```
Comment (dblock) [inline MIGRATION.md:6]: Since there's only one section, remove "Overview:"

File: MIGRATION.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+- Create a new Slack App as [advised by Slack](https://api.slack.com/legacy/custom-integrations/legacy-tokens#migrating-from-legacy).
```
Comment (dblock) [inline MIGRATION.md:7]: These are numbered, so 1. 2. 3. etc

Comment (dblock) [inline MIGRATION.md:7]: This is confusing. I think we should tell users to go through the tutorial/setup of slack-ruby-bot-server as in that doc, then copy command classes, then do the workflow that you suggest. And explain what happens when you do that and the fact that other teams can install the bot too then, which may not be what you want (and needs to be prevented).

Comment (wasabigeek) [reply MIGRATION.md]: I am actually a little confused by Slack itself. If I understood correctly, slack-ruby-bot-server requires the latest Slack Apps, but there are also legacy tokens AND class Slack Apps, both with subtly different migration strategies on the Slack end. Did I get that right, and should I gloss over this?

Comment (dblock) [reply MIGRATION.md]: I say gloss over this.

Comment (wasabigeek) [reply MIGRATION.md]: Ok! Any more comments?

**Comment by dblock** [comment]
I merged this, thank you.

**Pull request merged**